ecc/bls24-315/fr/pedersen/pedersen.go

```diff
@@ -212,34 +210,59 @@ func FoldCommitments(commitments []curve.G1Affine, fiatshamirSeeds ...[]byte) (c
 func (vk *VerifyingKey) Verify(commitment curve.G1Affine, knowledgeProof curve.G1Affine) error {
 
 	if !commitment.IsInSubGroup() || !knowledgeProof.IsInSubGroup() {
-		return fmt.Errorf("subgroup check failed")
+		return errors.New("subgroup check failed")
 	}
 
 	if isOne, err := curve.PairingCheck([]curve.G1Affine{commitment, knowledgeProof}, []curve.G2Affine{vk.G, vk.GRootSigmaNeg}); err != nil {
 		return err
 	} else if !isOne {
-		return fmt.Errorf("proof rejected")
+		return errors.New("proof rejected")
 	}
 	return nil
 }
 
-func getChallenge(fiatshamirSeeds [][]byte) (r fr.Element, err error) {
-	// incorporate user-provided seeds into the transcript
-	t := fiatshamir.NewTranscript(sha256.New(), "r")
-	for i := range fiatshamirSeeds {
-		if err = t.Bind("r", fiatshamirSeeds[i]); err != nil {
-			return
+// BatchVerify verifies n separately generated proofs of knowledge from different setup ceremonies, using n+1 pairings rather than 2n.
+func BatchVerify(vk []VerifyingKey, commitments []curve.G1Affine, pok []curve.G1Affine, challenge fr.Element) error {
```

Comment:
Change argument `challenge` to `combinationCoeff` and denote in method documentation that has to be random for security.

Also add, that the verification keys have to have same `G2` point.

Reply:
And you can make the `challenge`/`combinationCoeff` optional using an option. When it is not provided then we generate randomly inside the method.

Reply:
And, what do you think about renaming the method to `BatchVerifyMulti` indicating maybe better that we can use different verification keys (with same G2 though)